pkg/notary/registry.go

```diff
@@ -74,23 +65,6 @@ func (ir *imageResource) RegistryStr() string {
 	return ir.ref.Context().RegistryStr()
 }
 
-func getAuthenticator(ctx context.Context, ref string, registryClient images.Client) (*authn.Authenticator, error) {
-	parsedRef, err := name.ParseReference(ref)
-	if err != nil {
-		return nil, errors.Wrapf(err, "failed to parse registry reference %s", ref)
-	}
-
-	if err := registryClient.RefreshKeychainPullSecrets(ctx); err != nil {
```

Comment:
You can't remove this part, we need to refresh secrets to build a fresh keychain